Sources/Recycling/ListViews/FavoriteFolders/FavoriteFoldersListView.swift

```diff
@@ -0,0 +1,296 @@
+            searchBar.leadingAnchor.constraint(equalTo: leadingAnchor),
+            searchBar.trailingAnchor.constraint(equalTo: trailingAnchor),
+
+            tableView.topAnchor.constraint(equalTo: searchBar.bottomAnchor),
+            tableView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            tableView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            tableView.bottomAnchor.constraint(equalTo: bottomAnchor),
+
+            footerViewTop,
+            footerView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            footerView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            footerView.heightAnchor.constraint(equalToConstant: footerHeight)
+        ])
+    }
+}
+
+// MARK: - UITableViewDataSource
+
+extension FavoriteFoldersListView: UITableViewDataSource {
+    public func numberOfSections(in tableView: UITableView) -> Int {
+        return Section.allCases.count
+    }
+
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        guard let section = Section(rawValue: section) else { return 0 }
+
+        switch section {
+        case .addButton:
+            return isSearchActive ? 0 : 1
+        case .folders:
+            return dataSource?.numberOfItems(inFavoriteFoldersListView: self) ?? 0
+        }
+    }
+
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let section = Section(rawValue: indexPath.section) else { return UITableViewCell() }
+
+        switch section {
+        case .addButton:
+            let cell = tableView.dequeue(AddFavoriteFolderViewCell.self, for: indexPath)
+            cell.configure(withTitle: viewModel.addFolderText)
+            return cell
+        case .folders:
+            let cell = tableView.dequeue(RemoteImageTableViewCell.self, for: indexPath)
+
+            // Show a pretty color while we load the image
+            let colors: [UIColor] = [.toothPaste, .mint, .banana, .salmon]
+            let color = colors[indexPath.row % 4]
+
+            cell.loadingColor = color
+            cell.dataSource = self
+
+            if let viewModel = dataSource?.favoriteFoldersListView(self, viewModelAtIndex: indexPath.row) {
+                cell.configure(with: viewModel)
+            }
+
+            return cell
+        }
+    }
+}
+
+// MARK: - UITableViewDelegate
+
+extension FavoriteFoldersListView: UITableViewDelegate {
+    public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: false)
+
+        guard let section = Section(rawValue: indexPath.section) else { return }
+
+        switch section {
+        case .addButton:
+            delegate?.favoriteFoldersListViewDidSelectAddButton(self)
+        case .folders:
+            delegate?.favoriteFoldersListView(self, didSelectItemAtIndex: indexPath.row)
```

Comment:
Note to self _(since I'm supposed to make changes to this PR)_:

We should probably send the view model instead, seeing as the list of folders may be filtered, which the delegate probably doesn't know about.